<a href="https://colab.research.google.com/github/Umizmar/The_theory_of_probability/blob/main/%D0%A3%D1%80%D0%BE%D0%BA_6_%D0%A1%D1%80%D0%B0%D0%B2%D0%BD%D0%B5%D0%BD%D0%B8%D0%B5_%D0%B4%D0%BE%D0%BB%D0%B5%D0%B9_%D0%9F%D0%BE%D1%81%D1%82%D1%80%D0%BE%D0%B5%D0%BD%D0%B8%D0%B5_%D0%B4%D0%BE%D0%B2%D0%B5%D1%80%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D0%BE%D0%B3%D0%BE_%D0%B8%D0%BD%D1%82%D0%B5%D1%80%D0%B2%D0%B0%D0%BB%D0%B0.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

##Урок 6. Сравнение долей. Построение доверительного интервала

1. Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.


Поскольку известно среднее квадратичное генеральной совокупности, то
используем формулу с z критерием:
$$ \bar X \pm z_{\alpha/2}\cdot \frac {\sigma}{\sqrt n}$$

$\bar X = 80$  
z от 2.5% = 1.96 по таблице  
$\sigma=16$  
$n=256$

In [12]:
import math
print(80+1.96*(16/math.sqrt(256)))
80-1.96*(16/math.sqrt(256))

81.96


78.04

**Ответ**: [78.04; 81.96]

2. В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.

Поскольку в задаче имеется выборка малого объема, применим распределение Стьюдента.$$ \bar X \pm t_{\alpha/2}\cdot \frac {\sigma}{\sqrt n}$$
n=10

In [7]:
import scipy.stats as stats
import numpy as np

data=[6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1]
x = np.mean(data)
sigma = np.std(data, ddof=1)
print(f'Выборочное среднее-{x}\nСреднее квадратическое отклонение-{sigma}')

Выборочное среднее-6.590000000000001
Среднее квадратическое отклонение-0.4508017549014448


$\bar X=6.59$  
$\sigma = 0.45$

Найдем t-критерий

In [10]:
t=stats.t.ppf(0.975,len(data)-1)
t

2.2621571627409915

$$ 6.59 \pm 2.26\cdot \frac {0.45}{\sqrt 10}$$

In [14]:
print(6.59+2.226*(0.45/math.sqrt(10)))
6.59-2.226*(0.45/math.sqrt(10))

6.9067653532190665


6.273234646780933

**Ответ**: истинное значение случайной величины лежит в доверительном интервале (6.27; 6.91) с доверительной вероятностью 0,95.

 3. Рост дочерей 175, 167, 154, 174, 178, 148, 160, 167, 169, 170 Рост матерей  178, 165, 165, 173, 168, 155, 160, 164, 178, 175 Используя эти данные построить 95% доверительный интервал для разности среднего роста родителей и детей.

Интервальная оценка для разности средних арифметических  
$Δ \pm t_{\alpha/2} \cdot S_{Δ}$, где $Δ=\bar X_1-\bar X_2$

In [16]:
mother = [178, 165, 165, 173, 168, 155, 160, 164, 178, 175]
daughter = [175, 167, 154, 174, 178, 148, 160, 167, 169, 170]
x1=np.mean(mother)
x2=np.mean(daughter)
x= x1-x2
x

1.9000000000000057

$Δ=1.9$

Найдем t, где степень своды = 2*(n-1)=18

In [23]:
t=stats.t.ppf(0.975,18)
t

2.10092204024096

t=2.26  
$S_{Δ}=\sqrt {\frac{D}{n_1}+\frac{D}{n_2}}$, где n1 и n2 объемы выборок и = 10  
$D=\frac {1}{2}(D1+D2)$

In [19]:
D1=np.var(mother, ddof=1)
D2=np.var(daughter, ddof=1)
D=(D1+D2)/2
D

75.58333333333334

D=75.58

Найдем стандартную ошибку для разности двух средних арифметических $S_Δ$

In [22]:
s=np.sqrt(2*(75.58/10))
s

3.8879300405228485

Найдем интервал

In [26]:
L = 1.9 - (2.1*3.9)
U = 1.9 + (2.1*3.9)
L,U

(-6.289999999999999, 10.09)

**Ответ**: доверительный интервал для разности средних арифметических (-6.29;10.09)